### PR TITLE
PLANNER-2443: Fix Quarkus native: only check @PlanningSolution and @PlanningEntity members, make Substitute_ConfigUtil newInstance.signature match ConfigUtil.newInstance signature

### PR DESCRIPTION
**Pull request by Christopher-Chianelli**
I noticed the native build was broken when trying CR3; it appears the PLANNING_SCORE member was either copied or moved to ScoreDescriptor, causing code to be generated for it (although it a static field, which cause issues, thus native build failing)

### Checklist
- [ ] Documentation updated if applicable.
- [ ] Upgrade recipe provided if applicable.

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* for a <b>pull request build</b> please add comment: <b>Jenkins retest this</b>
* for a <b>full downstream build</b> please add comment: <b>Jenkins run fdb</b>
* for a <b>compile downstream build</b> please add comment: <b>Jenkins run cdb</b>
* for a <b>full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>
* for an <b>upstream build</b> please add comment: <b>Jenkins run upstream</b>
* for a <b>Quarkus LTS check</b> please add comment: <b>Jenkins run LTS</b>
* for a <b>Native check</b> please add comment: <b>Jenkins run native</b>
</details>

### CODE REVIEW

**Review by triceo — CHANGES_REQUESTED** [review]
I have to say I do not understand the problem, and the PR does not help with that.
There should be a test, which solves my problem, and also prevents regressions.

**Review by triceo — APPROVED** [review]
We discussed this and came up with a better implementation. LGTM.
Jenkins run native.

**Comment by Christopher-Chianelli** [comment]
Look like there is a config mismatch in (build) optaplanner:
```
Error when executing always post condition:

org.jenkinsci.plugins.workflow.steps.MissingContextVariableException: Required context class hudson.FilePath is missing
Perhaps you forgot to surround the code with a step that provides this, such as: node
```

**Comment by Christopher-Chianelli** [comment]
Error log; fails here: can confirm it is related to the PR (works on master, not on this PR)
```
String taskId = given()
                .contentType(ContentType.JSON)
                .when()
                .get("/rest/flights/" + id + "/tasks")
                .then()
                .statusCode(200)
                .body("$.size", is(1))
                .body("[0].name", is("finalizeSeatAssignment"))
                .extract()
                .path("[0].id");

```


```
[2021-06-07T16:44:56.366Z] [INFO] -------------------------------------------------------
[2021-06-07T16:44:56.366Z] [INFO]  T E S T S
[2021-06-07T16:44:56.366Z] [INFO] -------------------------------------------------------
[2021-06-07T16:44:57.305Z] [INFO] Running org.kie.kogito.examples.FlightTest
[2021-06-07T16:45:10.737Z] 2021-06-07 12:45:04,950 INFO  [org.kie.kog.cod.api.uti.AddonsConfigDiscovery] (build-4) Performed addonsConfig discovery, found: AddonsConfig{usePersistence=false, useTracing=false, useMonitoring=false, usePrometheusMonitoring=false, useCloudEvents=false, useExplainability=false, useProcessSVG=false}
[2021-06-07T16:45:10.737Z] 2021-06-07 12:45:05,281 INFO  [org.kie.kog.cod.cor.uti.ApplicationGeneratorDiscovery] (build-4) Generator discovery performed, found [openapispecs, processes, rules, decisions, predictions]
[2021-06-07T16:45:10.737Z] 2021-06-07 12:45:06,962 INFO  [org.kie.kog.cod.api.uti.AddonsConfigDiscovery] (build-34) Performed addonsConfig discovery, found: AddonsConfig{usePersistence=false, useTracing=false, useMonitoring=false, usePrometheusMonitoring=false, useCloudEvents=false, useExplainability=false, useProcessSVG=false}
[2021-06-07T16:45:10.737Z] 2021-06-07 12:45:06,970 INFO  [org.kie.kog.qua.com.dep.KogitoQuarkusResourceUtils] (build-34) No Java source to compile
[2021-06-07T16:45:10.737Z] 2021-06-07 12:45:08,791 INFO  [io.quarkus] (main) Quarkus 1.13.3.Final on JVM started in 4.889s. Listening on: http://localhost:37230
[2021-06-07T16:45:10.737Z] 2021-06-07 12:45:08,792 INFO  [io.quarkus] (main) Profile test activated. 
[2021-06-07T16:45:10.737Z] 2021-06-07 12:45:08,792 INFO  [io.quarkus] (main) Installed features: [cdi, kogito-decisions, kogito-predictions, kogito-processes, kogito-rules, optaplanner, optaplanner-jackson, resteasy, resteasy-jackson, servlet, smallrye-health, smallrye-openapi, swagger-ui]
[2021-06-07T16:45:12.115Z] [ERROR] WARNING: An illegal reflective access operation has occurred
[2021-06-07T16:45:12.115Z] [ERROR] WARNING: Illegal reflective access by org.codehaus.groovy.vmplugin.v9.Java9 (file:/home/jenkins/.m2/repository/org/codehaus/groovy/groovy/3.0.7/groovy-3.0.7.jar) to field java.util.ArrayList.size
[2021-06-07T16:45:12.115Z] [ERROR] WARNING: Please consider reporting this to the maintainers of org.codehaus.groovy.vmplugin.v9.Java9
[2021-06-07T16:45:12.115Z] [ERROR] WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
[2021-06-07T16:45:12.115Z] [ERROR] WARNING: All illegal access operations will be denied in a future release
[2021-06-07T16:45:24.356Z] [ERROR] Tests run: 1, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 25.462 s <<< FAILURE! - in org.kie.kogito.examples.FlightTest
[2021-06-07T16:45:24.356Z] [ERROR] org.kie.kogito.examples.FlightTest.testFlightsProcess  Time elapsed: 12.911 s  <<< FAILURE!
[2021-06-07T16:45:24.356Z] java.lang.AssertionError: 
[2021-06-07T16:45:24.356Z] 1 expectation failed.
[2021-06-07T16:45:24.356Z] Expected status code <200> but was <404>.
[2021-06-07T16:45:24.356Z] 
[2021-06-07T16:45:24.356Z] 	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
[2021-06-07T16:45:24.356Z] 	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
[2021-06-07T16:45:24.356Z] 	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
[2021-06-07T16:45:24.356Z] 	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
[2021-06-07T16:45:24.356Z] 	at org.codehaus.groovy.reflection.CachedConstructor.invoke(CachedConstructor.java:72)
[2021-06-07T16:45:24.356Z] 	at org.codehaus.groovy.reflection.CachedConstructor.doConstructorInvoke(CachedConstructor.java:59)
[2021-06-07T16:45:24.356Z] 	at org.codehaus.groovy.runtime.callsite.ConstructorSite$ConstructorSiteNoUnwrap.callConstructor(ConstructorSite.java:84)
[2021-06-07T16:45:24.356Z] 	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallConstructor(CallSiteArray.java:59)
[2021-06-07T16:45:24.356Z] 	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callConstructor(AbstractCallSite.java:263)
[2021-06-07T16:45:24.356Z] 	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callConstructor(AbstractCallSite.java:277)
[2021-06-07T16:45:24.356Z] 	at io.restassured.internal.ResponseSpecificationImpl$HamcrestAssertionClosure.validate(ResponseSpecificationImpl.groovy:493)
[2021-06-07T16:45:24.356Z] 	at io.restassured.internal.ResponseSpecificationImpl$HamcrestAssertionClosure$validate$1.call(Unknown Source)
[2021-06-07T16:45:24.356Z] 	at io.restassured.internal.ResponseSpecificationImpl.validateResponseIfRequired(ResponseSpecificationImpl.groovy:674)
[2021-06-07T16:45:24.356Z] 	at jdk.internal.reflect.GeneratedMethodAccessor146.invoke(Unknown Source)
[2021-06-07T16:45:24.356Z] 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[2021-06-07T16:45:24.356Z] 	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
[2021-06-07T16:45:24.356Z] 	at org.codehaus.groovy.runtime.callsite.PlainObjectMetaMethodSite.doInvoke(PlainObjectMetaMethodSite.java:43)
[2021-06-07T16:45:24.356Z] 	at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite$PogoCachedMethodSiteNoUnwrapNoCoerce.invoke(PogoMetaMethodSite.java:193)
[2021-06-07T16:45:24.356Z] 	at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite.callCurrent(PogoMetaMethodSite.java:61)
[2021-06-07T16:45:24.356Z] 	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:185)
[2021-06-07T16:45:24.356Z] 	at io.restassured.internal.ResponseSpecificationImpl.statusCode(ResponseSpecificationImpl.groovy:126)
[2021-06-07T16:45:24.356Z] 	at io.restassured.specification.ResponseSpecification$statusCode$0.callCurrent(Unknown Source)
[2021-06-07T16:45:24.356Z] 	at io.restassured.internal.ResponseSpecificationImpl.statusCode(ResponseSpecificationImpl.groovy:134)
[2021-06-07T16:45:24.356Z] 	at io.restassured.internal.ValidatableResponseOptionsImpl.statusCode(ValidatableResponseOptionsImpl.java:89)
[2021-06-07T16:45:24.356Z] 	at io.restassured.internal.ValidatableResponseImpl.super$2$statusCode(ValidatableResponseImpl.groovy)
[2021-06-07T16:45:24.356Z] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[2021-06-07T16:45:24.356Z] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[2021-06-07T16:45:24.356Z] 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[2021-06-07T16:45:24.356Z] 	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
[2021-06-07T16:45:24.356Z] 	at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:107)
[2021-06-07T16:45:24.356Z] 	at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:323)
[2021-06-07T16:45:24.356Z] 	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1268)
[2021-06-07T16:45:24.356Z] 	at org.codehaus.groovy.runtime.ScriptBytecodeAdapter.invokeMethodOnSuperN(ScriptBytecodeAdapter.java:144)
[2021-06-07T16:45:24.356Z] 	at io.restassured.internal.ValidatableResponseImpl.statusCode(ValidatableResponseImpl.groovy:142)
[2021-06-07T16:45:24.356Z] 	at io.restassured.internal.ValidatableResponseImpl.statusCode(ValidatableResponseImpl.groovy)
[2021-06-07T16:45:24.356Z] 	at org.kie.kogito.examples.FlightTest.testFlightsProcess(FlightTest.java:160)
[2021-06-07T16:45:24.356Z] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[2021-06-07T16:45:24.356Z] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[2021-06-07T16:45:24.356Z] 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[2021-06-07T16:45:24.356Z] 	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
[2021-06-07T16:45:24.356Z] 	at io.quarkus.test.junit.QuarkusTestExtension.runExtensionMethod(QuarkusTestExtension.java:944)
[2021-06-07T16:45:24.356Z] 	at io.quarkus.test.junit.QuarkusTestExtension.interceptTestMethod(QuarkusTestExtension.java:850)
[2021-06-07T16:45:24.356Z] 	at org.junit.jupiter.engine.execution.ExecutableInvoker$ReflectiveInterceptorCall.lambda$ofVoidMethod$0(ExecutableInvoker.java:115)
[2021-06-07T16:45:24.356Z] 	at org.junit.jupiter.engine.execution.ExecutableInvoker.lambda$invoke$0(ExecutableInvoker.java:105)
[2021-06-07T16:45:24.356Z] 	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$InterceptedInvocation.proceed(InvocationInterceptorChain.java:106)
[2021-06-07T16:45:24.356Z] 	at org.junit.jupiter.engine.extension.TimeoutExtension.intercept(TimeoutExtension.java:149)
[2021-06-07T16:45:24.356Z] 	at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestableMethod(TimeoutExtension.java:140)
[2021-06-07T16:45:24.356Z] 	at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestMethod(TimeoutExtension.java:84)
[2021-06-07T16:45:24.356Z] 	at org.junit.jupiter.engine.execution.ExecutableInvoker$ReflectiveInterceptorCall.lambda$ofVoidMethod$0(ExecutableInvoker.java:115)
[2021-06-07T16:45:24.356Z] 	at org.junit.jupiter.engine.execution.ExecutableInvoker.lambda$invoke$0(ExecutableInvoker.java:105)
[2021-06-07T16:45:24.356Z] 	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$InterceptedInvocation.proceed(InvocationInterceptorChain.java:106)
[2021-06-07T16:45:24.356Z] 	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.proceed(InvocationInterceptorChain.java:64)
[2021-06-07T16:45:24.356Z] 	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.chainAndInvoke(InvocationInterceptorChain.java:45)
[2021-06-07T16:45:24.356Z] 	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.invoke(InvocationInterceptorChain.java:37)
[2021-06-07T16:45:24.356Z] 	at org.junit.jupiter.engine.execution.ExecutableInvoker.invoke(ExecutableInvoker.java:104)
[2021-06-07T16:45:24.356Z] 	at org.junit.jupiter.engine.execution.ExecutableInvoker.invoke(ExecutableInvoker.java:98)
[2021-06-07T16:45:24.356Z] 	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.lambda$invokeTestMethod$6(TestMethodTestDescriptor.java:210)
[2021-06-07T16:45:24.356Z] 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
[2021-06-07T16:45:24.357Z] 	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.invokeTestMethod(TestMethodTestDescriptor.java:206)
[2021-06-07T16:45:24.357Z] 	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:131)
[2021-06-07T16:45:24.357Z] 	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:65)
[2021-06-07T16:45:24.357Z] 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:139)
[2021-06-07T16:45:24.357Z] 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
[2021-06-07T16:45:24.357Z] 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:129)
[2021-06-07T16:45:24.357Z] 	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
[2021-06-07T16:45:24.357Z] 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:127)
[2021-06-07T16:45:24.357Z] 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
[2021-06-07T16:45:24.357Z] 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:126)
[2021-06-07T16:45:24.357Z] 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:84)
[2021-06-07T16:45:24.357Z] 	at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
[2021-06-07T16:45:24.357Z] 	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:38)
[2021-06-07T16:45:24.357Z] 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:143)
[2021-06-07T16:45:24.357Z] 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
[2021-06-07T16:45:24.357Z] 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:129)
[2021-06-07T16:45:24.357Z] 	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
[2021-06-07T16:45:24.357Z] 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:127)
[2021-06-07T16:45:24.357Z] 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
[2021-06-07T16:45:24.357Z] 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:126)
[2021-06-07T16:45:24.357Z] 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:84)
[2021-06-07T16:45:24.357Z] 	at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
[2021-06-07T16:45:24.357Z] 	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:38)
[2021-06-07T16:45:24.357Z] 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:143)
[2021-06-07T16:45:24.357Z] 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
[2021-06-07T16:45:24.357Z] 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:129)
[2021-06-07T16:45:24.357Z] 	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
[2021-06-07T16:45:24.357Z] 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:127)
[2021-06-07T16:45:24.357Z] 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
[2021-06-07T16:45:24.357Z] 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:126)
[2021-06-07T16:45:24.357Z] 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:84)
[2021-06-07T16:45:24.357Z] 	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.submit(SameThreadHierarchicalTestExecutorService.java:32)
[2021-06-07T16:45:24.357Z] 	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor.execute(HierarchicalTestExecutor.java:57)
[2021-06-07T16:45:24.357Z] 	at org.junit.platform.engine.support.hierarchical.HierarchicalTestEngine.execute(HierarchicalTestEngine.java:51)
[2021-06-07T16:45:24.357Z] 	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:108)
[2021-06-07T16:45:24.357Z] 	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:88)
[2021-06-07T16:45:24.357Z] 	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.lambda$execute$0(EngineExecutionOrchestrator.java:54)
[2021-06-07T16:45:24.357Z] 	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.withInterceptedStreams(EngineExecutionOrchestrator.java:67)
[2021-06-07T16:45:24.357Z] 	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:52)
[2021-06-07T16:45:24.357Z] 	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:96)
[2021-06-07T16:45:24.357Z] 	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:75)
[2021-06-07T16:45:24.357Z] 	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.execute(JUnitPlatformProvider.java:188)
[2021-06-07T16:45:24.357Z] 	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invokeAllTests(JUnitPlatformProvider.java:154)
[2021-06-07T16:45:24.357Z] 	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invoke(JUnitPlatformProvider.java:128)
[2021-06-07T16:45:24.357Z] 	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:428)
[2021-06-07T16:45:24.357Z] 	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:162)
[2021-06-07T16:45:24.357Z] 	at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:562)
[2021-06-07T16:45:24.357Z] 	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:548)
[2021-06-07T16:45:24.357Z] 
[2021-06-07T16:45:24.357Z] 2021-06-07 12:45:21,800 INFO  [io.quarkus] (main) Quarkus stopped in 0.053s
[2021-06-07T16:45:24.357Z] [INFO] 
[2021-06-07T16:45:24.357Z] [INFO] Results:
[2021-06-07T16:45:24.357Z] [INFO] 
[2021-06-07T16:45:24.357Z] [ERROR] Failures: 
[2021-06-07T16:45:24.357Z] [ERROR]   FlightTest.testFlightsProcess:160 1 expectation failed.
[2021-06-07T16:45:24.357Z] Expected status code <200> but was <404>.
[2021-06-07T16:45:24.357Z] 
[2021-06-07T16:45:24.357Z] [INFO] 
[2021-06-07T16:45:24.357Z] [ERROR] Tests run: 1, Failures: 1, Errors: 0, Skipped: 0
[2021-06-07T16:45:24.357Z] [INFO] 
```

**Comment by Christopher-Chianelli** [comment]
Found the flaw: ConstraintConfiguration are not `@PlanningEntity`/`@PlanningSolution`, but have member accessors

**Comment by Christopher-Chianelli** [comment]
(an additional flaw that make this approach unfeasible: `@PlanningId` can be put on Problem facts, which are not annotated and does not need to extend/implement anything; changing this to a static field check approach; if the field is not static, then the bytecode generated is valid.)

**Comment by triceo** [comment]
Jenkins run native.

File: optaplanner-quarkus-integration/optaplanner-quarkus/deployment/src/main/java/org/optaplanner/quarkus/deployment/OptaPlannerProcessor.java [hunk 1]
```diff
@@ -551,42 +537,39 @@ private GeneratedGizmoClasses generateDomainAccessors(SolverConfig solverConfig,
...
-                            }
+                        try {
+                            generatedMemberAccessorsClassNameSet
+                                    .add(GizmoMemberAccessorEntityEnhancer.generateFieldAccessor(annotatedMember, indexView,
+                                            classOutput,
+                                            classInfo, fieldInfo, transformers));
+                        } catch (ClassNotFoundException | NoSuchFieldException e) {
+                            throw new IllegalStateException("Fail to generate member accessor for field (" +
+                                    fieldInfo.name() + ") of class " +
+                                    classInfo.name().toString() + ".", e);
```
Comment (ge0ffrey) [inline optaplanner-quarkus-integration/optaplanner-quarkus/deployment/src/main/java/org/optaplanner/quarkus/deployment/OptaPlannerProcessor.java:556]: To make it consistent with the error message at line 571
```suggestion
                                    fieldInfo.name() + ") of the class( " +
                                    classInfo.name().toString() + ").", e);
```

**Review by ge0ffrey — APPROVED** [review]

**Comment by triceo** [comment]
Jenkins retest this.
Jenkins run native.

**Comment by triceo** [comment]
(Oops, clicked the wrong button.)

**Comment by triceo** [comment]
@Christopher-Chianelli It appears that native is still broken. Please check it out.

**Comment by Christopher-Chianelli** [comment]
@triceo fixed, but I basically needed to make one of the tests not test anything (the issue was GraalVM does not like calling constructors on abstract classes/interfaces, so I don't generate constructors for them. But the test solverConfig is filled with abstract classes/interfaces, which means I cannot verify they are added to the bean factory (since they are skipped as they are abstract)

**Comment by Christopher-Chianelli** [comment]
Jenkins run native.

**Comment by triceo** [comment]
@Christopher-Chianelli In that case, let's fix the test properly as opposed to introducing technical debt. Please address to the TODO there.

**Comment by Christopher-Chianelli** [comment]
@triceo done

**Comment by triceo** [comment]
Good! Unfortunately native still appears to be failing - I'm not in the VPN anymore, so I can not check if the failures are real.

**Comment by Christopher-Chianelli** [comment]
@triceo wgeb I checked, it was a failure in kogito-examples explaniability, which doesn't use optaplanner; the other native checks are still running IIRC

**Comment by Christopher-Chianelli** [comment]
@triceo look like it another failure, although the other failure been fixed; Lesson does not have a public no-args constructor, which cause an error

**Comment by Christopher-Chianelli** [comment]
I can't exactly automatically generate a constructor for classes, since there might be finals

**Comment by triceo** [comment]
I guess I'm more curious as to why is this suddenly a problem. It seems to me that we have not changed anything related.

**Comment by Christopher-Chianelli** [comment]
@triceo cause we are using GIZMO by default now; it was using REFLECTION a long time before

**Comment by Christopher-Chianelli** [comment]
I think this is more of an error on the example than the code (since we need a no-args constructor that we can access for cloning) (although I can make it report a better error message). It less of an issue in the JVM because we can setAccessible everything. Native naturally doesn't like that.

**Comment by triceo** [comment]
I think it makes sense to require a no-arg constructor for cloning. Therefore as long as the error message explains what is happening and how to fix it, we should be fine. 

Although I am curious how the normal non-Quarkus code handles this. There are the following possibilities:

- There is no constructor. Therefore, Java will auto-generate a public no-args constructor.
- There is a no-args constructor, but it is not accessible. (Private, for example.) In that case, we can use setAccessible(), although that is ugly.
- There is a constructor, but it has arguments. In that case, Java does not auto-generate anything. Yet the cloner still needs to work. How does it do that?

The Gizmo behavior should be as similar as possible.

**Comment by Christopher-Chianelli** [comment]
- There is no constructor. Therefore, Java will auto-generate a public no-args constructor.
  Gizmo behaves the same as Java in this case

- There is a no-args constructor, but it is not accessible. (Private, for example.) In that case, we can use setAccessible(), although that is ugly.
 This is the case that cause the crash. It works for reflection thanks to
 setAccessible, Gizmo doesn't do that so it wouldn't work. What I might be able
to do is change its accessibility to public via a Class Vistor

- There is a constructor, but it has arguments. In that case, Java does not auto-generate anything. Yet the cloner still needs to work. How does it do that?
  Java also crash in this case:
```
            try {
                constructor = clazz.getDeclaredConstructor();
            } catch (ReflectiveOperationException e) {
                throw new IllegalStateException("The class (" + clazz
                        + ") should have a no-arg constructor to create a planning clone.", e);
            }
```

**Comment by triceo** [comment]
OK, in that case, it is relatively simple. We have the first and third case covered already.
In the second case, let's try to make the constructor accessible. If it fails (it can also fail in Java), let's fail with a reasonable error message.

**Comment by triceo** [comment]
And all of this behavior really needs to be tested.
(Plus we need to enable the native CI by default, but that's for another time.)

**Comment by Christopher-Chianelli** [comment]
@triceo done.
Jenkins run native.

**Comment by triceo** [comment]
kogito-apps failure in explainability may be related. Rerunning to make sure.

Jenkins retest this.
Jenkins run native.

**Comment by Christopher-Chianelli** [comment]
@triceo looking at the logs, it does not seem related. Errors are related to `drools-reflective` for apps (which we do not use in quarkus), and the examples build hanged during a test in Process Usertasks Security OIDC Keycloak Quarkus

**Comment by triceo** [comment]
Ok, two out of three it is. :-)

Jenkins run native.
Jenkinks retest this.

**Comment by Christopher-Chianelli** [comment]
@triceo I am 99% sure the native failure is unrelated to this PR; the error message was the same as the last run, with no mention of optaplanner anywhere in the stack traces; seems to do with drools-reflective. Considering that native builds already fail without this PR, and this PR make more native build pass, and that this PR is well tested, is there anything blocking merging this PR?

**Pull request merged**